### PR TITLE
Add a "Glossary" page

### LINKED ISSUE #34: Add a "Glossary" page
This can be used together with the solution to https://github.com/Australian-Genomics/CTRL/issues/19 to allow administrators to create and link to glossaries.

### PR DESCRIPTION
**Pull request by DoxasticFox**
Resolves #34 and #29.

Demo:

[Peek 2022-08-16 18-08.webm](https://user-images.githubusercontent.com/11529449/184830545-22d06406-c944-4e37-beae-cc654e2fcd4e.webm)

Admin portal:

Screenshot_20220816_181056

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
+unless StudyCode.find_by(title: 'A1543457')
+  StudyCode.create!(title: 'A1543457')
+end
+
```
Comment (DoxasticFox) [inline db/seeds.rb:4]: This is the part which resolves https://github.com/Australian-Genomics/CTRL/issues/29. I know it should be its own PR, but it's just four teeny-tiny lines...

**Comment by DoxasticFox** [comment]
Feedback from @MatildaAHaas and @rosiejbrown: The "Documentation" page in the admin portal should describe how to create and link to glossary entries.

**Review by mathob — COMMENTED** [review]
This is a good idea. I was able to add a glossary term and additional it as a hyperlink in a consent question.
- can user interaction with a glossary term be implemented as a pop up showing the definition on mouseover, rather than a a link to the glossary page
- currently links within areas other than consent questions (e.g. the description filed of a consent step) are not rendered

**Comment by DoxasticFox** [comment]
>can user interaction with a glossary term be implemented as a pop up showing the definition on mouseover, rather than a a link to the glossary page

@mathob I think we'd probably want the definition to show on-click instead of on-hover for the sake of mobile users, but otherwise, I think your suggestion would improve the UX. Do you have suggestions for how the administrator should specify which phrase in the survey maps to which definition? In the current design, the user specifies what phrase links to which glossary definition using HTML `<a>` tags ([example](https://github.com/Australian-Genomics/CTRL/blob/cf88c2a200e6fa0d3a5f9ae05166e42fe424f9d7/db/seed_data/step_two_questions.yml#L4)). Given that the definitions wouldn't be links anymore, I think it might be most appropriate to use the `<dfn id="term-id">...</dfn>` tags and hook them up as pop-ups using JavaScript magic. SGTY?

I think it might also be worth getting @MatildaAHaas's and @rosiejbrown's input on the UX seeing as they seemed happy with the current implementation when I demoed it to them on Wednesday, and they asked me if I could have #20 and #24 done soon for the sake of a demo. So maybe it'd be best to merge this PR as-is and raise another issue to re-work it.

>currently links within areas other than consent questions (e.g. the description filed of a consent step) are not rendered

I noticed that this was a problem for the description on the first survey step, but I think the others are all fine. I'll double-check...

**Comment by DoxasticFox** [comment]
>currently links within areas other than consent questions (e.g. the description filed of a consent step) are not rendered

I've double-checked and I believe that they are rendered now. Hyperlinks were an issue on the first step prior to https://github.com/Australian-Genomics/CTRL/pull/35 being merged. But it's merged now and I've rebased on `main`, so hyperlinks should work on this branch now. Please let me know if it still isn't working

**Review by rosiejbrown — APPROVED** [review]

**Review by mathob — APPROVED** [review]
